tests/merger_test.py

```diff
@@ -294,33 +269,16 @@ def test_merge_point_data_unique_entity_name_unique_name(tmp_path):
         nan_array = np.empty(10)
         nan_array[:] = np.nan
 
-        # sort the dictionary by its keys
-        merged_data = list(
-            dict(
-                sorted(
-                    {
-                        child.name: child.values
-                        for child in test.children
-                        if isinstance(child, Data)
-                    }.items()
-                )
-            ).values()
-        )
-
-        np.testing.assert_almost_equal(
-            merged_data[0], np.hstack((nan_array, data[3].values))
-        )
-
         np.testing.assert_almost_equal(
-            merged_data[1], np.hstack((data[0].values, nan_array))
```

Comment:
What if the data with similar entity_type and  name are in the second object (or the nth object?
)